### ISSUE TITLE
Peer review Samuel Hempelt

### ISSUE THREAD

**Issue by hempelt** [MarioKrupinski/Data-Analytics#2]
### Peer review 

- Peer review by: Samuel Hempelt

### Ziel des Projektes
Ziel des Projektes ist es die die Anzahl der erwarteten Hauptuntersuchungen in den nächsten 30 Tagen für verschiedene Werkstätte vorhersagen, um eine bessere Planung für Umsatz und Personal zu gewährleisten. Die Vorhersage könnte ein gutes Feature für die Software der GTÜ (Gesellschaft für technische Überwachung) sein und hat damit das Potential weitere Werkstätten als Partner für die GTÜ zu gewinnen.

### Daten
Der importierte Dataframe enthält folgende Informationen:
•	Datum: Datum, an dem die HU durchgeführt wurde
•	HU: Anzahl der HUs, die an diesem Tag in dieser Werkstatt/Unterbuero durchgeführt wurden
•	Bundesland: Bundesland, in dem die Werkstatt/Unterbuero ist
•	Unique Unterbüro Anonym: Anonymsierter Name der Werkstatt/Unterbüros

Die Daten wurden transformiert und bereinigt. Außerdem werden eine Reihe von zusätzlichen Variablen berechnet:
•	Der Datentyp des Attributs „DATUM“ wurde von einem string-Datentyp in ein Date-Format konvertiert
•	Der Dataframe wurde ergänzt, sodass jedes Unterbüro für jeden Tag eine eigene Zeile besitzt
•	Inaktive Phasen, in denen das Unterbüro keine HUs durchführt, wurden aus dem Dataframe entfernt
•	Aus dem Attribut „DATUM“ wurden weitere unabhängige Variablen berechnet und dem Dataframe hinzugefügt: „WOCHENTAG“, „TAG_IM_MONAT“, „MONAT“, „TAGE_SEIT_START“
•	Durch die Aggregation mehrerer Datensätze wurden weitere Variablen dem Dataframe hinzugefügt: Beispielsweise den Mittelwert aller Montage in einem Monat („WOCHENTAG_ROLLING_MEAN”)  oder der Mittelwert für die letzten 30 Tage pro Gruppe (“AVG_HU_LETZE_30_TAGE”)
•	Zusätzlich wurden sogenannte Lag-Werte dem Dataframe hinzugefügt: Der Lag_1 gibt den HU-Wert des Vortages an, Lag_2 den HU-Wert von vor 2 Tagen usw.
•	Die Variablen IS_FEIERTAG" und "IS_FERIEN" wurden angelegt, um den Einfluss von Feiertagen und Ferien im Model zu berücksichtigen

### Approach, tools and methods

-	Pandas Funktionen zur Formatierung, Aggregation, Filtern, Hinzufügen von Variablen und zur Darstellung des Dataframes
-	Plotly zur Visualisierung der Daten in der Daten Exploration 
-	Numpy zur Berechnung des Interquartilsabstands
-	Pandas Corr-Funktion zur Berechnung der Korrelationen
-	Mit Plotly und seaborn zur Erstellung der Heatmap für die Visualisierung der Korrelationen in einer Korrelationsmatrix
-	Berechnung der Mutual Information mit der sklearn.mutural_info_regression-Funktion 
-	Lightgbm Funktionen für die Entwicklung eines LightGBM Modells

### Lack of clarity
•	An manchen Stellen könnte man auf die Darstellung des gesamten Codes verzichten. Beispielsweise verhindert das Hinzufügen der Feiertage und Ferien eine gute Übersicht über die Arbeit. Man könnte diese Abschnitte zum Beispiel in den Anhang packen.
•	Aus irgendeinem Grund lagen in Github die Tabellen nicht formatiert sondern als ungerenderter Markdown code vor. Nachdem ich sie heruntergeladen und in Visual Studio code geöffnet hatte, hat es aber funktioniert.
•	In Code Zelle 21 erscheint ein Fehler: “name  ‘px’ is not defined. Wahrscheinlich weil plotly erst an späterer Stelle (Codezelle 24 importiert wurde)

### Possible improvements
•	Einige Abschnitte zur Bereinigung und Erweiterung des Dataframes nehmen sehr viel Raum ein. Man könnte diese Abschnitte in ein separates Dokument auslagern, sodass der Fokus der Arbeit auf der Datenanalyse und der Entwicklung des Models bestehen bleibt.
•	Eine ausführlichere Beschreibung fortgeschrittener Methoden und Vorgehensweisen wäre hilfreich, da sie nicht jedem bekannt sind. (Gradient Boosting, Mutual Information, Hyperparameteranpassung)
•	Die Visualisierung der Vorhersage ist ungünstig gewählt, da die Vorhersage nur für einen eingeschränkten Zeitraum vorlag, jedoch der gesamte Zeitraum visualisiert wurde (Anzahl aller HUs pro Tag und Vorhersage)

### Presentation
•	Wie du in der Conclusion erwähnst, wäre eine Gegenüberstellung mehrerer Modelle interessant.
•	Bei der Präsentation würde ich alle Data Engineering Schritte nur nebenbei erwähnen
•.      Eine Feature Analyse fände ich sehr interessant
•	Super finde ich die Visualisierungen der Vorhersagen. Eine Anpassung der X-Achse oder eine Erweiterung der vorhergesagten Tage wäre hier für eine bessere Darstellung hilfreich.

### Further comments
Die Arbeit ist sehr umfangreich und es werden viele fortgeschrittene Methoden verwendet, die ich bisher noch nicht kannte. Durch die vielen Data Engineering Zwischenschritte ist es mir schwer gefallen die Übersicht zu behalten. Dennoch konnte ich viel Lernen und werde einige Ideen wie die Suche nach Ausreißern oder die Visualisierung der Vorhersagen auch für die eigene Arbeit übernehmen.
